src/dns_utils.rs

```diff
@@ -49,21 +51,18 @@ pub fn parse_dns_answer(msg: &[u8]) -> Result<()> {
 
 /// Fetches `odohconfig` by querying the target server for HTTPS records.
 /// Currently supports `odoh.cloudflare-dns.com.`.
-pub fn fetch_odoh_config(target: &str) -> Result<Vec<u8>> {
+pub async fn fetch_odoh_config(target: &str) -> Result<Vec<u8>> {
     target
         .find(CONFIG_DOMAIN)
         .ok_or_else(|| anyhow!("Target not supported"))?;
     let address = CONFIG_RESOLVER.parse()?;
-    let conn = UdpClientConnection::new(address)?;
-    let client = SyncClient::new(conn);
+    let conn = UdpClientStream::<UdpSocket>::new(address);
+    let (mut client, bg) = AsyncClient::connect(conn).await?;
+    tokio::spawn(async move { bg.await });
```

Comment:
This is going to run the connection driver in the background indefinitely. You can either use something like https://docs.rs/futures/0.3.1/futures/future/fn.select.html to await on both `client.query` + driver, or spawn a task with a handle that's going to return as soon as this function finishes (like https://docs.rs/tokio/0.2.23/tokio/sync/oneshot/fn.channel.html).

Reply:
Fixed